test/unit/org/apache/cassandra/journal/IndexTest.java

```diff
@@ -0,0 +1,263 @@
+
+        File directory = new File(Files.createTempDirectory(null));
+        directory.deleteOnExit();
+        Descriptor descriptor = Descriptor.create(directory, System.currentTimeMillis(), 1);
+        inMemory.persist(descriptor);
+
+        try (OnDiskIndex<TimeUUID> onDisk = OnDiskIndex.open(descriptor, TimeUUIDKeySupport.INSTANCE))
+        {
+            assertArrayEquals(EMPTY, onDisk.lookUp(key0));
+            assertArrayEquals(new int[] { val11 }, onDisk.lookUp(key1));
+            assertArrayEquals(new int[] { val21, val22 }, onDisk.lookUp(key2));
+            assertArrayEquals(new int[] { val31, val32, val33 }, onDisk.lookUp(key3));
+            assertArrayEquals(EMPTY, onDisk.lookUp(key4));
+
+            assertEquals(key1, onDisk.firstId());
+            assertEquals(key3, onDisk.lastId());
+
+            assertFalse(onDisk.mayContainId(key0));
+             assertTrue(onDisk.mayContainId(key1));
+             assertTrue(onDisk.mayContainId(key2));
+             assertTrue(onDisk.mayContainId(key3));
+            assertFalse(onDisk.mayContainId(key4));
+        }
+    }
+
+    @Test
+    public void prop() throws IOException
+    {
+        Constraint sizeConstraint = Constraint.between(1, 10);
+        Constraint valueSizeConstraint = Constraint.between(0, 10);
+        Constraint positionConstraint = Constraint.between(0, Integer.MAX_VALUE);
+        Gen<TimeUUID> keyGen = Generators.timeUUID();
+        Gen<int[]> valueGen = rs -> {
+            int[] array = new int[(int) rs.next(valueSizeConstraint)];
+            IntHashSet uniq = new IntHashSet();
+            for (int i = 0; i < array.length; i++)
+            {
+                int value = (int) rs.next(positionConstraint);
+                while (!uniq.add(value))
+                    value = (int) rs.next(positionConstraint);
+                array[i] = value;
+            }
+            return array;
+        };
+        Gen<Map<TimeUUID, int[]>> gen = rs -> {
+            int size = (int) rs.next(sizeConstraint);
+            Map<TimeUUID, int[]> map = Maps.newHashMapWithExpectedSize(size);
+            for (int i = 0; i < size; i++)
+            {
+                TimeUUID key = keyGen.generate(rs);
+                while (map.containsKey(key))
+                    key = keyGen.generate(rs);
+                int[] value = valueGen.generate(rs);
+                map.put(key, value);
+            }
+            return map;
+        };
+        gen = gen.describedAs(map -> {
+            StringBuilder sb = new StringBuilder();
+            for (Map.Entry<TimeUUID, int[]> entry : map.entrySet())
+                sb.append('\n').append(entry.getKey()).append('\t').append(Arrays.toString(entry.getValue()));
+            return sb.toString();
+        });
+        File directory = new File(Files.createTempDirectory(null));
+        directory.deleteOnExit();
+        qt().withFixedSeed(1735898489413583L).forAll(gen).checkAssert(map -> test(directory, map));
```

Comment:
should remove `.withFixedSeed(1735898489413583L)`

Reply:
Done